### ISSUE TITLE
Fix dependency lower bounds

### ISSUE THREAD

**Issue by andreasabel** [jacobstanley/unix-compat#49]
matrix.hackage reports build failures of 0.5.3 under GHC 7.4 and 7.0:
<img width="742" alt="Screenshot 2021-11-20 at 19 40 18" src="https://user-images.githubusercontent.com/1155218/142737558-c0ae73fa-988f-4a40-a596-e1bbd19d3137.png">
    GHC-7.4.2/unix-compat-0.5.3

E.g. https://matrix.hackage.haskell.org/#/package/unix-compat/0.5.3/ghc-7.4.2@1636831695
```
Building library for unix-compat-0.5.3..
...
[7 of 8] Compiling System.PosixCompat.Files ( /tmp/matrix-worker/1611227319/dist-newstyle/build/x86_64-linux/ghc-7.4.2/unix-compat-0.5.3/build/System/PosixCompat/Files.hs, /tmp/matrix-worker/1611227319/dist-newstyle/build/x86_64-linux/ghc-7.4.2/unix-compat-0.5.3/build/System/PosixCompat/Files.o )

src/System/PosixCompat/Files.hsc:62:7:
    Not in scope: `accessTimeHiRes'

src/System/PosixCompat/Files.hsc:63:7:
    Not in scope: `modificationTimeHiRes'
    Perhaps you meant `modificationTime' (imported from System.Posix.Files)

src/System/PosixCompat/Files.hsc:64:7:
    Not in scope: `statusChangeTimeHiRes'
    Perhaps you meant `statusChangeTime' (imported from System.Posix.Files)
<<ghc: 151407976 bytes, 163 GCs, 4346654/10212592 avg/max bytes residency (5 samples), 20M in use, 0.00 INIT (0.00 elapsed), 0.09 MUT (0.15 elapsed), 0.08 GC (0.08 elapsed) :ghc>>
```

**Comment by andreasabel** [comment]
@jacobstanley Thanks for your work on `unix-compat` and the new version `0.5.4`.
Unfortunately, the newly released 0.5.4 still suffers from the problem I reported already on 0.5.3.  
https://github.com/gregwebs/Shelly.hs/runs/5681270882?check_suite_focus=true
Build plan:
```
2022-03-24T18:33:53.4732952Z  └─ unix-compat-0.5.4
2022-03-24T18:33:53.4733214Z     ├─ base-4.5.1.0 ┄┄
2022-03-24T18:33:53.4733479Z     └─ unix-2.5.1.1 ┄┄
```
Build attempt:
```
2022-03-24T18:36:30.1476486Z Building library for unix-compat-0.5.4..
2022-03-24T18:36:30.1480669Z [7 of 8] Compiling System.PosixCompat.Files ( dist/build/System/PosixCompat/Files.hs, dist/build/System/PosixCompat/Files.o )
2022-03-24T18:36:30.1480954Z 
2022-03-24T18:36:30.1481097Z src/System/PosixCompat/Files.hsc:62:7:
2022-03-24T18:36:30.1481424Z     Not in scope: `accessTimeHiRes'
2022-03-24T18:36:30.1481777Z 
2022-03-24T18:36:30.1481877Z src/System/PosixCompat/Files.hsc:63:7:
2022-03-24T18:36:30.1482427Z     Not in scope: `modificationTimeHiRes'
2022-03-24T18:36:30.1482975Z     Perhaps you meant `modificationTime' (imported from System.Posix.Files)
2022-03-24T18:36:30.1483610Z 
2022-03-24T18:36:30.1483776Z src/System/PosixCompat/Files.hsc:64:7:
2022-03-24T18:36:30.1484248Z     Not in scope: `statusChangeTimeHiRes'
2022-03-24T18:36:30.1485036Z     Perhaps you meant `statusChangeTime' (imported from System.Posix.Files)
```
According to https://hackage.haskell.org/package/unix-2.7.2.2/changelog, the unknown functions were added in 2.6.0.0:
> New functions accessTimeHiRes, modificationTimeHiRes, and statusChangeTimeHiRes for accessing high resolution timestamps

Consequently, the lower bound `>=2.4` on `unix` is wrong: https://github.com/jacobstanley/unix-compat/blob/0cef494ca8d3345b97fa121545c57a33210c6c56/unix-compat.cabal#L67
Increasing it to `>=2.6` should solve the problem.

**Comment by jacobstanley** [comment]
> I made the respective revisions on hackage.

Thanks! ❤️

I've merged your PR.
